packages/block-library/src/template-part/edit/placeholder/index.js

```diff
@@ -21,21 +26,19 @@ export default function TemplatePartPlaceholder( {
 	const { saveEntityRecord } = useDispatch( 'core' );
 	const onCreate = useCallback( async () => {
 		const title = 'Untitled Template Part';
-		const slug = cleanForSlug( title );
+		const slug = cleanForSlug( title ) + '-' + uuid();
```

Comment:
What's the point of appending this?

Reply:
This is probably something that should be done in the server somehow, the slug needs to be unique.

Reply:
Doesn‘t WP already ensure uniqueness?

Reply:
not in my tests at least, maybe there's something specific done for these CPTs. as we only need uniqueness per theme.

Reply:
it does seem odd needing to handle this ad hoc

Reply:
Bit of a drive-by while I'm still reading up on context for a more in-depth review of this PR, so forgive me if this is a red herring, but could it be because of this?

https://github.com/WordPress/gutenberg/blob/dc52d16dd0469cde9725441015515b1406e19ec7/lib/template-parts.php#L62-L80

Reply:
Definitely related. Thanks @ockham I think instead of bypassing entirely, we need to ensure it's unique within the same theme.

Reply:
I moved the unique slug generation to the server in 9f792dc